pages/api/user/[id].ts

```diff
@@ -33,7 +33,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (req.method === "DELETE") {
-    return res.status(405).json({ message: "Method Not Allowed" });
+    await prisma.user.delete({
+      where: {
+        id: authenticatedUser?.id,
+      },
+    });
+    return res.status(200).json({ message: "User Deleted" });
```

Comment:
I prefer (as you return no details about the entity just deleted, so you'd never use this response message):
```javascript
res.status(204).end()
```